server/src/main/java/org/opensearch/common/util/concurrent/OpenSearchExecutors.java

```diff
@@ -137,6 +137,19 @@ public static OpenSearchThreadPoolExecutor newFixed(String name, int size, int q
             queue, threadFactory, new OpenSearchAbortPolicy(), contextHolder);
     }
 
+    public static OpenSearchThreadPoolExecutor newResizable(String name, int size, int queueCapacity,
+                                                    ThreadFactory threadFactory, ThreadContext contextHolder) {
+        BlockingQueue<Runnable> queue;
+        if (queueCapacity < 0) {
+            queue = ConcurrentCollections.newBlockingQueue();
+        } else {
+            queue = new SifiResizableBlockingQueue<>(ConcurrentCollections.<Runnable>newBlockingQueue(),
```

Comment:
```suggestion
            queue = new OpenSearchResizableBlockingQueue<>(ConcurrentCollections.<Runnable>newBlockingQueue(),
```

Reply:
thanks. made the changed as per your suggestion